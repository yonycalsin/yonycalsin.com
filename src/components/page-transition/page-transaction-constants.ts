const VARIANTS = {
  initial: {
    opacity: 0,
    y: 8,
  },
  enter: {
    opacity: 1,
    y: 0,
    transition: {
      duration: 0.4,
      ease: [0.61, 1, 0.88, 1],
    },
  },
}

export { VARIANTS }
