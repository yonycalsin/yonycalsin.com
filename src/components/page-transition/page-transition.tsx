import * as React from 'react'
import { motion } from 'framer-motion'

import type { PageTransitionProps } from 'typings/components'
import { VARIANTS } from './page-transaction-constants'

function PageTransition(props: PageTransitionProps) {
  const { children } = props

  return (
    <motion.div initial="initial" animate="enter" variants={VARIANTS}>
      {children}
    </motion.div>
  )
}

export default PageTransition
