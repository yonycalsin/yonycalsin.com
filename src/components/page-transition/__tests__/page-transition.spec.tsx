import { render } from '@testing-library/react'

import { PageTransition } from '../page-transition'

describe('Components / Page Transaction', () => {
  it('wrappers a component with page-transaction correctly', () => {
    const view = render(
      <PageTransition>
        <h1>Content</h1>
      </PageTransition>,
    )

    expect(view.baseElement).toMatchSnapshot()
  })
})
