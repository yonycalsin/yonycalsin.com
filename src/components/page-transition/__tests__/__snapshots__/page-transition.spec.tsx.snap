// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTransaction wrappers a component with page-transaction correctly 1`] = `
<body>
  <div>
    <div
      style="opacity: 0; transform: translateY(8px) translateZ(0);"
    >
      <div
        data-testid="mock-content"
      />
    </div>
  </div>
</body>
`;
